### PR TITLE
Minor changes to grid column

### LINKED ISSUE #892: Make minor changes to existing plugins
All these plugins are part of the group OKF v2:
---

- [ ] **Article Link**
    - [ ] Add on hover / on click states (search for "Article" [here](https://ishigami.github.io/okfn_front/components.html))
    - [ ] Make URL field non-mandatory

---

- [ ] **Banner**
    - [ ] Implement different colors for each variation (search for "Banner" [here](https://ishigami.github.io/okfn_front/components.html))
    - [ ] Enable markdown in the `Text` field
---

- [ ] **Carousel**
    - [ ] Display each image's Alt text after 1 sec mouse hover
---

- [ ] OKImage
    - [ ] When not `In column`, display the caption below the image, not inside 
    - [ ] Change the option text from `Show caption` to `Use caption instead of text`
    - [ ] Display Alt text after 1 sec mouse hover

---
- [ ] Grid Column
    - [ ] How to "skip" a column? Example: say I want to add 2 blocks using the 2 central columns in a grid of 4. Suggestion: Maybe make it work like `Pills Menu` that always has the same size starting from the center
---

- [ ] Feature Block
    - [ ] Add the small arrow to the "Learn more" button (gets activated when an URL is added)
    - [ ] `Yellow` and `Yellow background` currently have the same layout. Needs to recover `Yellow` as the original yellow squared box (search for "Content Grid" [here](https://ishigami.github.io/okfn_front/components.html))
    - [ ] Add option `Text white` to `Blue background`, `Yellow background`, `Purple background` and `Rounded corners`
    - [ ] Add on hover / on click states to `Blue background`, `Yellow background`, `Purple background` and `Rounded corners`
    - [ ] Change type name from `Yellow` to `Yellow Box`
    - [ ] Change variant name from `White` to `White Box`

---

- [ ] Footer
    - [ ] Replace link of `Request a quote` with `/contact`

---

- [ ] Full Banner
    - [ ] Add option `Text White` (including a white button)

---

- [ ] Gallery
    - [ ] Rename plugin to "Running Gallery"
    - [ ] Stop automatic scroll on mouse hover
    - [ ] Make it draggable
    - [ ] Display each image's Alt text after 1 sec mouse hover
    
---

- [ ] Heading
    - [ ] Enable markdown in the `Title` field

---

- [ ] List
    - [ ] Rename option from `XL` to `XL no title`

---

- [ ] Newsletter
    - [ ] Rename plugin to "Call-to-action"
    - [ ] Create field to add button link
    - [ ] Prefill the linked form with the email added by the user
    - [ ] For type `Small` - Make the heading editable
    - [ ] For type `Small` - Decrease text and image sizes, align elements (search for "Campaign > Small" [here](https://ishigami.github.io/okfn_front/components-2.html))

---

- [ ] Page Header
    - [ ] Rename field from `Second text` to `Highlight text`
    - [ ] For type `Image left / Text right` - Centralise
    - [ ] For type `Image left / Text right` - Add bottom margin (the element is too close to the next one)

---

- [ ] PillButton
    - [ ] For type `Arrow` - I can't see the arrow, please add (search for "Button > Arrow" [here](https://ishigami.github.io/okfn_front/components.html))

---

- [ ] Quote
    - [ ] Enable markdown in the `Text` field

---

- [ ] Spotlight
    - [ ] Display Alt text after 1 sec mouse hover

### PR DESCRIPTION
**Pull request by shashigharti**
- Partial fixes #892

Grid Column

- [x] How to "skip" a column? Example: say I want to add 2 blocks using the 2 central columns in a grid of 4. Suggestion: Maybe make it work like Pills Menu that always has the same size starting from the center

### CODE REVIEW

File: foundation/okfplugins/grid_columns/templates/grid_columns_plugin.html [hunk 1]
```diff
@@ -1,10 +1,14 @@
 {% load cms_tags %}
 <div class="sect -tighten-1 sect-generic mb-20">
     <div class="sect__content">
-      <div class="grid {{ grid_cols_class }} gap-10 md:gap-3">
-      {% for plugin in instance.child_plugin_instances %}
-          {% render_plugin plugin %}
-      {% endfor %}
+      <div class="grid md:grid-cols-{{ instance.columns }} gap-10 md:gap-3">
```
Comment (pdelboca) [inline foundation/okfplugins/grid_columns/templates/grid_columns_plugin.html:4]: Tailwind does not get along well with Dynamic Classes (https://tailwindcss.com/docs/content-configuration#dynamic-class-names) so we should avoid using interpolation.

That's why I created the `grid_cols_class` variable in the view.

**Review by pdelboca — APPROVED** [review]
@shashigharti  approved!

### Some notes:

As the comment line describes, this looks like a quick fix for something that might not be clearly defined in the requirements. Maybe we are torturing `grids` to get something that could be done otherwise?

Also look at my comment about Dynamic CSS classes.

@lucpretti what's the end goal of this feature? Which kind of layout are you trying to make?

@lucpretti  take into consideration that this will only skip X columns and it will not center the elements like the pills buttons since grid layouts are designed to keep stricted elements. 

So adding 4 columns, 1 skip and only two elements will result in huge margings and a small gap:
image

**Comment by shashigharti** [comment]
Thank you! @pdelboca . I have restored the code.

**Comment by lucpretti** [comment]
> @lucpretti what's the end goal of this feature? Which kind of layout are you trying to make?
> 
> @lucpretti take into consideration that this will only skip X columns and it will not center the elements like the pills buttons since grid layouts are designed to keep stricted elements.
> 
> So adding 4 columns, 1 skip and only two elements will result in huge margings and a small gap: image

@pdelboca @shashigharti The goal is to keep the boxes' vertical layout centered (just like in your screenshot above) instead of displaying them in wider boxes, as currently occurs if the grid is set to "2" (see below).

Screenshot 2023-08-21 at 12 53 49

I think you guys solved the issue already, don't you? Is it deployed to staging?

**Pull request merged**